chimney/src/main/scala/io/scalaland/chimney/internal/macros/Model.scala

```diff
@@ -20,6 +20,19 @@ trait Model extends TransformerConfiguration {
   }
 
   case class TransformerBodyTree(tree: Tree, isWrapped: Boolean)
-  case class ResolvedAccessor(symbol: MethodSymbol, wasRenamed: Boolean)
 
+  sealed trait AccessorResolution extends Product with Serializable {
+    def isResolved: Boolean
+  }
+  object AccessorResolution {
+    case object NotFound extends AccessorResolution {
+      override val isResolved: Boolean = false
```

Comment:
Isn't it better to keep it (and other `isResolved`) as `def`?

Reply:
I think it's quite normal to define something as `def` in trait and override it using `val`. I don't see any advantage of using `def` here

Reply:
`val` causes immediate memory allocation, even if it's not called. The way that we use in `resolveAccessorsMapping` suggests that for common use case a lot of `NotFound` objects are produced, for which `isResolved` has no chance to be called.

Not a big thing, but I think making it `def` would make it more GC friendly.

Reply:
👌